api/src/main/java/ca/bc/gov/educ/api/distribution/util/EducDistributionApiUtils.java

```diff
@@ -216,9 +218,60 @@ public static void zipFile(File fileToZip, String fileName, ZipOutputStream zipO
 			while ((length = fis.read(bytes)) >= 0) {
 				zipOut.write(bytes, 0, length);
 			}
-		}catch (Exception e) {
+			zipOut.closeEntry();
```

Comment:
zipOut.closeEntry() should be in finally block

Reply:
In case of the Exception output stream will get EOF anyway and zip file will be corrupted. No really make sense to put it in the final block

Reply:
Should always close a stream. Since zipoutputstream implements closeable, you could also use try-with-resources, etc.

Reply:
zipOut.closeEntry()  doesn't close the stream. As you see this function is recursive. Make sense to close the stream in the caller writeZipFile function

Reply:
Wait... don't merge please. Moving the zipOut.close() to the final block corrupts the  zip file